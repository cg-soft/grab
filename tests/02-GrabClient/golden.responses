
GET http://host:port/any?id=any&op=config
{
  "data": {
    "debug": true,
    "gc_interval": "1000000",
    "hash": "",
    "help": false,
    "host": "127.0.0.1",
    "port": "port",
    "timeout": "1000000",
    "timestamp": timestamp,
    "version": "VERSION"
  },
  "id": "any/any",
  "op": "config",
  "resource": "/any",
  "status": "ok"
}
== After config any by any:
GET http://host:port/?id=any&op=dump
{
  "data": {
    "config": {
      "debug": true,
      "gc_interval": "1000000",
      "hash": "",
      "help": false,
      "host": "127.0.0.1",
      "port": "port",
      "timeout": "1000000",
      "timestamp": timestamp,
      "version": "VERSION"
    },
    "queues": {},
    "stats": {
      "by_lock": {},
      "by_owner": {},
      "max_queue_length": 0,
      "max_queue_owner": null,
      "max_queue_resource": null,
      "total": 0,
      "uptime": 0
    }
  },
  "id": "any/",
  "op": "dump",
  "resource": "/",
  "status": "ok"
}
==
grab me/lunch/1
GET http://host:port/lunch/1?id=me&op=grab
{
  "data": {
    "/lunch": 1,
    "/lunch/1": 1
  },
  "id": "me/lunch/1",
  "op": "grab",
  "resource": "/lunch/1",
  "status": "ok"
}
== After grab lunch/1 by me:
GET http://host:port/?id=me&op=dump
{
  "data": {
    "config": {
      "debug": true,
      "gc_interval": "1000000",
      "hash": "",
      "help": false,
      "host": "127.0.0.1",
      "port": "port",
      "timeout": "1000000",
      "timestamp": timestamp,
      "version": "VERSION"
    },
    "queues": {
      "/lunch": [
        {
          "exes": {},
          "owners": {
            "me/lunch/1": timestamp
          },
          "shared": true
        }
      ],
      "/lunch/1": [
        {
          "exes": {},
          "owners": {
            "me/lunch/1": timestamp
          },
          "shared": false
        }
      ]
    },
    "stats": {
      "by_lock": {
        "/lunch": 1,
        "/lunch/1": 1
      },
      "by_owner": {
        "me/lunch/1": 2
      },
      "max_queue_length": 1,
      "max_queue_owner": "me/lunch/1",
      "max_queue_resource": "/lunch/1",
      "total": 2,
      "uptime": 0
    }
  },
  "id": "me/",
  "op": "dump",
  "resource": "/",
  "status": "ok"
}
==
grab you/lunch/2
GET http://host:port/lunch/2?id=you&op=grab
{
  "data": {
    "/lunch": 1,
    "/lunch/2": 1
  },
  "id": "you/lunch/2",
  "op": "grab",
  "resource": "/lunch/2",
  "status": "ok"
}
== After grab lunch/2 by you:
GET http://host:port/?id=you&op=dump
{
  "data": {
    "config": {
      "debug": true,
      "gc_interval": "1000000",
      "hash": "",
      "help": false,
      "host": "127.0.0.1",
      "port": "port",
      "timeout": "1000000",
      "timestamp": timestamp,
      "version": "VERSION"
    },
    "queues": {
      "/lunch": [
        {
          "exes": {},
          "owners": {
            "me/lunch/1": timestamp,
            "you/lunch/2": timestamp
          },
          "shared": true
        }
      ],
      "/lunch/1": [
        {
          "exes": {},
          "owners": {
            "me/lunch/1": timestamp
          },
          "shared": false
        }
      ],
      "/lunch/2": [
        {
          "exes": {},
          "owners": {
            "you/lunch/2": timestamp
          },
          "shared": false
        }
      ]
    },
    "stats": {
      "by_lock": {
        "/lunch": 2,
        "/lunch/1": 1,
        "/lunch/2": 1
      },
      "by_owner": {
        "me/lunch/1": 2,
        "you/lunch/2": 2
      },
      "max_queue_length": 1,
      "max_queue_owner": "me/lunch/1",
      "max_queue_resource": "/lunch/1",
      "total": 4,
      "uptime": 0
    }
  },
  "id": "you/",
  "op": "dump",
  "resource": "/",
  "status": "ok"
}
==
release me/lunch/2
GET http://host:port/lunch/2?id=you&op=release
{
  "data": 1,
  "id": "you/lunch/2",
  "op": "release",
  "resource": "/lunch/2",
  "status": "ok"
}
== After release lunch/2 by you:
GET http://host:port/?id=you&op=dump
{
  "data": {
    "config": {
      "debug": true,
      "gc_interval": "1000000",
      "hash": "",
      "help": false,
      "host": "127.0.0.1",
      "port": "port",
      "timeout": "1000000",
      "timestamp": timestamp,
      "version": "VERSION"
    },
    "queues": {
      "/lunch": [
        {
          "exes": {
            "you/lunch/2": timestamp
          },
          "owners": {
            "me/lunch/1": timestamp
          },
          "shared": true
        }
      ],
      "/lunch/1": [
        {
          "exes": {},
          "owners": {
            "me/lunch/1": timestamp
          },
          "shared": false
        }
      ]
    },
    "stats": {
      "by_lock": {
        "/lunch": 2,
        "/lunch/1": 1,
        "/lunch/2": 1
      },
      "by_owner": {
        "me/lunch/1": 2,
        "you/lunch/2": 2
      },
      "max_queue_length": 1,
      "max_queue_owner": "me/lunch/1",
      "max_queue_resource": "/lunch/1",
      "total": 4,
      "uptime": 0
    }
  },
  "id": "you/",
  "op": "dump",
  "resource": "/",
  "status": "ok"
}
==
release me/lunch/1
GET http://host:port/lunch/1?id=me&op=release
{
  "data": 2,
  "id": "me/lunch/1",
  "op": "release",
  "resource": "/lunch/1",
  "status": "ok"
}
== After release lunch/1 by me:
GET http://host:port/?id=me&op=dump
{
  "data": {
    "config": {
      "debug": true,
      "gc_interval": "1000000",
      "hash": "",
      "help": false,
      "host": "127.0.0.1",
      "port": "port",
      "timeout": "1000000",
      "timestamp": timestamp,
      "version": "VERSION"
    },
    "queues": {},
    "stats": {
      "by_lock": {
        "/lunch": 2,
        "/lunch/1": 1,
        "/lunch/2": 1
      },
      "by_owner": {
        "me/lunch/1": 2,
        "you/lunch/2": 2
      },
      "max_queue_length": 1,
      "max_queue_owner": "me/lunch/1",
      "max_queue_resource": "/lunch/1",
      "total": 4,
      "uptime": 0
    }
  },
  "id": "me/",
  "op": "dump",
  "resource": "/",
  "status": "ok"
}
==
stats
GET http://host:port/all?id=any&op=stats
{
  "data": {
    "by_lock": {
      "/lunch": 2,
      "/lunch/1": 1,
      "/lunch/2": 1
    },
    "by_owner": {
      "me/lunch/1": 2,
      "you/lunch/2": 2
    },
    "max_queue_length": 1,
    "max_queue_owner": "me/lunch/1",
    "max_queue_resource": "/lunch/1",
    "total": 4,
    "uptime": uptime
  },
  "id": "any/all",
  "op": "stats",
  "resource": "/all",
  "status": "ok"
}
== After stats all by any:
GET http://host:port/?id=any&op=dump
{
  "data": {
    "config": {
      "debug": true,
      "gc_interval": "1000000",
      "hash": "",
      "help": false,
      "host": "127.0.0.1",
      "port": "port",
      "timeout": "1000000",
      "timestamp": timestamp,
      "version": "VERSION"
    },
    "queues": {},
    "stats": {
      "by_lock": {
        "/lunch": 2,
        "/lunch/1": 1,
        "/lunch/2": 1
      },
      "by_owner": {
        "me/lunch/1": 2,
        "you/lunch/2": 2
      },
      "max_queue_length": 1,
      "max_queue_owner": "me/lunch/1",
      "max_queue_resource": "/lunch/1",
      "total": 4,
      "uptime": uptime
    }
  },
  "id": "any/",
  "op": "dump",
  "resource": "/",
  "status": "ok"
}
==
shutdown should now work
GET http://host:port/?id=any&op=config
GET http://host:port/all?id=any&op=shutdown
{
  "data": 0,
  "id": "any/all",
  "op": "shutdown",
  "resource": "/all",
  "status": "ok"
}
== After shutdown all by any:
GET http://host:port/?id=any&op=dump
Failed to urlopen http://host:port/?id=any&op=dump
None
==
== Now with our own json.py ==

GET http://host:port/any?id=any&op=config
{
  "data": {
    "debug": true,
    "gc_interval": "1000000",
    "hash": "",
    "help": false,
    "host": "127.0.0.1",
    "port": "port",
    "timeout": "1000000",
    "timestamp": timestamp,
    "version": "VERSION"
  },
  "id": "any/any",
  "op": "config",
  "resource": "/any",
  "status": "ok"
}
== After config any by any:
GET http://host:port/?id=any&op=dump
{
  "data": {
    "config": {
      "debug": true,
      "gc_interval": "1000000",
      "hash": "",
      "help": false,
      "host": "127.0.0.1",
      "port": "port",
      "timeout": "1000000",
      "timestamp": timestamp,
      "version": "VERSION"
    },
    "queues": {},
    "stats": {
      "by_lock": {},
      "by_owner": {},
      "max_queue_length": 0,
      "max_queue_owner": null,
      "max_queue_resource": null,
      "total": 0,
      "uptime": 0
    }
  },
  "id": "any/",
  "op": "dump",
  "resource": "/",
  "status": "ok"
}
==
grab me/lunch/1
GET http://host:port/lunch/1?id=me&op=grab
{
  "data": {
    "/lunch": 1,
    "/lunch/1": 1
  },
  "id": "me/lunch/1",
  "op": "grab",
  "resource": "/lunch/1",
  "status": "ok"
}
== After grab lunch/1 by me:
GET http://host:port/?id=me&op=dump
{
  "data": {
    "config": {
      "debug": true,
      "gc_interval": "1000000",
      "hash": "",
      "help": false,
      "host": "127.0.0.1",
      "port": "port",
      "timeout": "1000000",
      "timestamp": timestamp,
      "version": "VERSION"
    },
    "queues": {
      "/lunch": [
        {
          "exes": {},
          "owners": {
            "me/lunch/1": timestamp
          },
          "shared": true
        }
      ],
      "/lunch/1": [
        {
          "exes": {},
          "owners": {
            "me/lunch/1": timestamp
          },
          "shared": false
        }
      ]
    },
    "stats": {
      "by_lock": {
        "/lunch": 1,
        "/lunch/1": 1
      },
      "by_owner": {
        "me/lunch/1": 2
      },
      "max_queue_length": 1,
      "max_queue_owner": "me/lunch/1",
      "max_queue_resource": "/lunch/1",
      "total": 2,
      "uptime": 0
    }
  },
  "id": "me/",
  "op": "dump",
  "resource": "/",
  "status": "ok"
}
==
grab you/lunch/2
GET http://host:port/lunch/2?id=you&op=grab
{
  "data": {
    "/lunch": 1,
    "/lunch/2": 1
  },
  "id": "you/lunch/2",
  "op": "grab",
  "resource": "/lunch/2",
  "status": "ok"
}
== After grab lunch/2 by you:
GET http://host:port/?id=you&op=dump
{
  "data": {
    "config": {
      "debug": true,
      "gc_interval": "1000000",
      "hash": "",
      "help": false,
      "host": "127.0.0.1",
      "port": "port",
      "timeout": "1000000",
      "timestamp": timestamp,
      "version": "VERSION"
    },
    "queues": {
      "/lunch": [
        {
          "exes": {},
          "owners": {
            "me/lunch/1": timestamp,
            "you/lunch/2": timestamp
          },
          "shared": true
        }
      ],
      "/lunch/1": [
        {
          "exes": {},
          "owners": {
            "me/lunch/1": timestamp
          },
          "shared": false
        }
      ],
      "/lunch/2": [
        {
          "exes": {},
          "owners": {
            "you/lunch/2": timestamp
          },
          "shared": false
        }
      ]
    },
    "stats": {
      "by_lock": {
        "/lunch": 2,
        "/lunch/1": 1,
        "/lunch/2": 1
      },
      "by_owner": {
        "me/lunch/1": 2,
        "you/lunch/2": 2
      },
      "max_queue_length": 1,
      "max_queue_owner": "me/lunch/1",
      "max_queue_resource": "/lunch/1",
      "total": 4,
      "uptime": 0
    }
  },
  "id": "you/",
  "op": "dump",
  "resource": "/",
  "status": "ok"
}
==
release me/lunch/2
GET http://host:port/lunch/2?id=you&op=release
{
  "data": 1,
  "id": "you/lunch/2",
  "op": "release",
  "resource": "/lunch/2",
  "status": "ok"
}
== After release lunch/2 by you:
GET http://host:port/?id=you&op=dump
{
  "data": {
    "config": {
      "debug": true,
      "gc_interval": "1000000",
      "hash": "",
      "help": false,
      "host": "127.0.0.1",
      "port": "port",
      "timeout": "1000000",
      "timestamp": timestamp,
      "version": "VERSION"
    },
    "queues": {
      "/lunch": [
        {
          "exes": {
            "you/lunch/2": timestamp
          },
          "owners": {
            "me/lunch/1": timestamp
          },
          "shared": true
        }
      ],
      "/lunch/1": [
        {
          "exes": {},
          "owners": {
            "me/lunch/1": timestamp
          },
          "shared": false
        }
      ]
    },
    "stats": {
      "by_lock": {
        "/lunch": 2,
        "/lunch/1": 1,
        "/lunch/2": 1
      },
      "by_owner": {
        "me/lunch/1": 2,
        "you/lunch/2": 2
      },
      "max_queue_length": 1,
      "max_queue_owner": "me/lunch/1",
      "max_queue_resource": "/lunch/1",
      "total": 4,
      "uptime": 0
    }
  },
  "id": "you/",
  "op": "dump",
  "resource": "/",
  "status": "ok"
}
==
release me/lunch/1
GET http://host:port/lunch/1?id=me&op=release
{
  "data": 2,
  "id": "me/lunch/1",
  "op": "release",
  "resource": "/lunch/1",
  "status": "ok"
}
== After release lunch/1 by me:
GET http://host:port/?id=me&op=dump
{
  "data": {
    "config": {
      "debug": true,
      "gc_interval": "1000000",
      "hash": "",
      "help": false,
      "host": "127.0.0.1",
      "port": "port",
      "timeout": "1000000",
      "timestamp": timestamp,
      "version": "VERSION"
    },
    "queues": {},
    "stats": {
      "by_lock": {
        "/lunch": 2,
        "/lunch/1": 1,
        "/lunch/2": 1
      },
      "by_owner": {
        "me/lunch/1": 2,
        "you/lunch/2": 2
      },
      "max_queue_length": 1,
      "max_queue_owner": "me/lunch/1",
      "max_queue_resource": "/lunch/1",
      "total": 4,
      "uptime": 0
    }
  },
  "id": "me/",
  "op": "dump",
  "resource": "/",
  "status": "ok"
}
==
stats
GET http://host:port/all?id=any&op=stats
{
  "data": {
    "by_lock": {
      "/lunch": 2,
      "/lunch/1": 1,
      "/lunch/2": 1
    },
    "by_owner": {
      "me/lunch/1": 2,
      "you/lunch/2": 2
    },
    "max_queue_length": 1,
    "max_queue_owner": "me/lunch/1",
    "max_queue_resource": "/lunch/1",
    "total": 4,
    "uptime": uptime
  },
  "id": "any/all",
  "op": "stats",
  "resource": "/all",
  "status": "ok"
}
== After stats all by any:
GET http://host:port/?id=any&op=dump
{
  "data": {
    "config": {
      "debug": true,
      "gc_interval": "1000000",
      "hash": "",
      "help": false,
      "host": "127.0.0.1",
      "port": "port",
      "timeout": "1000000",
      "timestamp": timestamp,
      "version": "VERSION"
    },
    "queues": {},
    "stats": {
      "by_lock": {
        "/lunch": 2,
        "/lunch/1": 1,
        "/lunch/2": 1
      },
      "by_owner": {
        "me/lunch/1": 2,
        "you/lunch/2": 2
      },
      "max_queue_length": 1,
      "max_queue_owner": "me/lunch/1",
      "max_queue_resource": "/lunch/1",
      "total": 4,
      "uptime": uptime
    }
  },
  "id": "any/",
  "op": "dump",
  "resource": "/",
  "status": "ok"
}
==
shutdown should now work
GET http://host:port/?id=any&op=config
GET http://host:port/all?id=any&op=shutdown
{
  "data": 0,
  "id": "any/all",
  "op": "shutdown",
  "resource": "/all",
  "status": "ok"
}
== After shutdown all by any:
GET http://host:port/?id=any&op=dump
Failed to urlopen http://host:port/?id=any&op=dump
None
==
