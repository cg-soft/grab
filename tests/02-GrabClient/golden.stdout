Server running at http://127.0.0.1:port
GET: op=config; id=any; until=timestamp; resource=/any
GET: op=dump; id=any; until=timestamp; resource=/
GET: op=grab; id=me; until=timestamp; resource=/lunch/1
Adding me/lunch/1 to end of /lunch/1
Adding me/lunch/1 to end of /lunch
GET: op=dump; id=me; until=timestamp; resource=/
GET: op=grab; id=you; until=timestamp; resource=/lunch/2
Adding you/lunch/2 to end of /lunch/2
Merging you/lunch/2 at end of /lunch
GET: op=dump; id=you; until=timestamp; resource=/
GET: op=release; id=you; until=timestamp; resource=/lunch/2
Found you/lunch/2 at /lunch/2[0]
Removing you/lunch/2 from /lunch/2
Found you/lunch/2 at /lunch[0]
GET: op=dump; id=you; until=timestamp; resource=/
GET: op=release; id=me; until=timestamp; resource=/lunch/1
Found me/lunch/1 at /lunch/1[0]
Removing me/lunch/1 from /lunch/1
Found me/lunch/1 at /lunch[0]
Removing you/lunch/2,me/lunch/1 from /lunch
GET: op=dump; id=me; until=timestamp; resource=/
GET: op=stats; id=any; until=timestamp; resource=/all
GET: op=dump; id=any; until=timestamp; resource=/
GET: op=config; id=any; until=timestamp; resource=/
GET: op=shutdown; id=any; until=timestamp; resource=/all
Shutting down
Server running at http://127.0.0.1:port
GET: op=config; id=any; until=timestamp; resource=/any
GET: op=dump; id=any; until=timestamp; resource=/
GET: op=grab; id=me; until=timestamp; resource=/lunch/1
Adding me/lunch/1 to end of /lunch/1
Adding me/lunch/1 to end of /lunch
GET: op=dump; id=me; until=timestamp; resource=/
GET: op=grab; id=you; until=timestamp; resource=/lunch/2
Adding you/lunch/2 to end of /lunch/2
Merging you/lunch/2 at end of /lunch
GET: op=dump; id=you; until=timestamp; resource=/
GET: op=release; id=you; until=timestamp; resource=/lunch/2
Found you/lunch/2 at /lunch/2[0]
Removing you/lunch/2 from /lunch/2
Found you/lunch/2 at /lunch[0]
GET: op=dump; id=you; until=timestamp; resource=/
GET: op=release; id=me; until=timestamp; resource=/lunch/1
Found me/lunch/1 at /lunch/1[0]
Removing me/lunch/1 from /lunch/1
Found me/lunch/1 at /lunch[0]
Removing you/lunch/2,me/lunch/1 from /lunch
GET: op=dump; id=me; until=timestamp; resource=/
GET: op=stats; id=any; until=timestamp; resource=/all
GET: op=dump; id=any; until=timestamp; resource=/
GET: op=config; id=any; until=timestamp; resource=/
GET: op=shutdown; id=any; until=timestamp; resource=/all
Shutting down
